src/MultiSelect/Input.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import propTypes from 'prop-types'
+import cx from 'classnames'
+import { multiSelectedPropType } from '../common-prop-types.js'
+import { colors } from '../theme.js'
+import { SelectionList } from './SelectionList.js'
+import { InputPlaceholder } from '../Select/InputPlaceholder.js'
+import { InputPrefix } from '../Select/InputPrefix.js'
+import { InputClearButton } from '../Select/InputClearButton.js'
+
+const Input = ({
+    selected,
+    onChange,
+    clearable,
+    clearText,
+    placeholder,
+    prefix,
+    className,
+    disabled,
+}) => {
+    const hasSelection = selected.length > 0
+    const onClear = e => {
+        e.stopPropagation()
+        onChange([])
+    }
+
+    return (
+        <div className={cx('root', className)}>
+            <InputPrefix prefix={prefix} />
+            {!hasSelection && !prefix && (
+                <InputPlaceholder placeholder={placeholder} />
+            )}
+            {hasSelection && (
+                <div>
```

Comment:
Why not move this one to the `SelectionList`?

Reply:
We could, but since we already have `hasSelection` in this context I thought it made sense to just cancel it early.

Reply:
I meant the `div`

Reply:
Ah. That's because it's there to enforce a wrapper for the chips so they wrap correctly on overflow (so it's related to the input and its flex layout). I didn't move it to the selectionlist because then it would be out of context and people might think it's superfluous and remove it, breaking the overflow wrapping of the input.

I've added a comment in 448918ae0107b8a44225a7529853ca26d445b3f6 (comment is below because otherwise I apparently would have to add a fragment).